### PR TITLE
Introduce a command launcher in the tests

### PR DESCRIPTION
**Pull request by damien-carcel**
When running the tests, we can indicates through the configuration parameter `pim.is_docker` if the PIM we run the tests against is installed locally or run through Docker.

We had several places in the test code where we did the same check "if docker then run this, else run that".

This PR introduce a new `CommandLauncher`, to be used everywhere in the tests. This launcher is the only piece of code in the tests to be aware if we run against Docker or not.

### CODE REVIEW

File: tests/DatabaseInstaller.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+     */
+    public function __construct(array $config)
+    {
+        $this->commandLauncher = new CommandLauncher($config);
+    }
+
+    /**
+     * Installs the PIM database.
+     *
+     * @throws ProcessFailedException If an error occurred during the PIM installation
```
Comment (willy-ahva) [inline tests/DatabaseInstaller.php]: nope

File: tests/CredentialGenerator.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    {
+        $this->commandLauncher = new CommandLauncher($config);
+    }
+
+    /**
+     * Generates credentials on the PIM.
+     *
+     * @param string $pimVersion Akeneo PIM version
+     *
+     * @throws ProcessFailedException If an error occurred during the generation process
```
Comment (willy-ahva) [inline tests/CredentialGenerator.php]: nope

File: tests/CredentialGenerator.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+{
+    /** @var CommandLauncher */
+    private $commandLauncher;
+
+    /**
+     * @param array $config
+     */
+    public function __construct(array $config)
+    {
+        $this->commandLauncher = new CommandLauncher($config);
```
Comment (ahocquard) [inline tests/CredentialGenerator.php]: inject command launcher instead of config

File: tests/CommandLauncher.php [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+ * Launches a command through a Symfony process.
+ *
+ * The launcher will decide if the command is to be launched locally or in a
+ * Docker container depending to the configuration passed to the constructor.
+ *
+ * @author    Damien Carcel <damien.carcel@gmail.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class CommandLauncher
```
Comment (ahocquard) [inline tests/CommandLauncher.php]: `ConsoleCommandLauncher` as it does not exec a just a shell command

File: tests/DatabaseInstaller.php [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+{
+    /** @var CommandLauncher */
+    private $commandLauncher;
+
+    /**
+     * @param array $config
+     */
+    public function __construct(array $config)
+    {
+        $this->commandLauncher = new CommandLauncher($config);
```
Comment (ahocquard) [inline tests/DatabaseInstaller.php]: same, inject the command launcher

File: composer.json [hunk 6]
```diff
@@ -11,7 +11,11 @@
   ],
   "autoload": {
     "psr-4": {
-      "Akeneo\\Pim\\": "src/",
+      "Akeneo\\Pim\\": "src/"
+    }
+  },
+  "autoload-dev": {
```
Comment (ahocquard) [inline composer.json]: 👍

File: tests/CredentialGenerator.php (hunk 3 above)
Comment (damien-carcel) [reply tests/CredentialGenerator.php]: Done, dunno why this preview was not updated.

File: tests/Common/Api/ApiTestCase.php [hunk 7]
```diff
@@ -68,12 +71,28 @@ protected function createClient()
...
+        return $this->configuration;
+    }
+
     /**
      * @throws \RuntimeException
      *
      * @return array
      */
-    protected function getConfiguration()
+    protected function parseConfigurationFile()
```
Comment (ahocquard) [inline tests/Common/Api/ApiTestCase.php]: private would even be better

**Pull request merged**